core/store/src/trie/receipts_column_helper.rs

```diff
@@ -0,0 +1,182 @@
+use crate::{get, set, TrieAccess, TrieUpdate};
+use near_primitives::errors::StorageError;
+use near_primitives::receipt::{DelayedReceiptIndices, Receipt};
+use near_primitives::trie_key::TrieKey;
+
+/// Type safe access to delayed receipts queue stored in the state. Only use one
+/// at the time for the same queue!
+///
+/// The struct keeps a in-memory copy of the queue indics to avoid reading it
+/// from the trie on every access. Modification are written back to the
+/// TrieUpdate immediately on every update.
+///
+/// But if you load two instance of this type at the same time, modifications on
+/// onw won't be synced to the other!
+pub struct DelayedReceiptQueue {
+    indices: DelayedReceiptIndices,
+}
+
+/// Read-only iterator over receipt queues stored in the state trie.
+///
+/// This iterator currently only supports delayed receipts but is already
+/// written general to work with the new queues that are going to be added for
+/// congestion control.
+pub struct ReceiptIterator<'a> {
+    trie_keys: Box<dyn Iterator<Item = TrieKey>>,
+    trie: &'a dyn TrieAccess,
+}
```

Comment:
mini nit: I would put that as the first thing in this file as the most generic one. Also to keep DelayedReceiptQueue definition and implementation together.

Reply:
done